Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -554,6 +554,23 @@ def create_pull_request(cls, repository, args, branch_point, callback=None, unbl
         push_env = os.environ.copy()
         push_env["VERBOSITY"] = str(args.verbose)
 
+        if target.endswith('fork') and repository.config().get('webkitscmpy.update-fork', 'false') == 'true':
+            # If our remote is a GitHub repository, we can use the API and save ourselves a push
+            fork_remote = repository.remote(name=target)
+            if isinstance(fork_remote, remote.GitHub):
```

Comment:
if not, should we fallback to rebase?

Reply:
It's not quite that we fallback to rebase, it's that we only have a fall back if we're rebasing.

Previous logic was basically "if you just updated your local version of the target branch (ie, rebased), then we should update your fork's version of that same branch because it's likely out of date"

Now, our logic is "If we can update your fork via an API call, do that, and then the remote's update". The previous logic of "if we didn't rebase, we have nothing new to push" still applies, if we don't have an API available to update our fork.